src/main/java/com/aws/greengrass/lifecyclemanager/GenericExternalService.java

```diff
@@ -120,9 +120,9 @@ protected GenericExternalService(Topics c, Topics privateSpace, Platform platfor
                 updateSystemResourceLimits();
             }
 
-            // Reinstall for changes to the install script or if the package version changed, or posixUser has changed
+            // Reinstall for changes to the install script or if the package version changed, or runWith user changed
             if (child.childOf(Lifecycle.LIFECYCLE_INSTALL_NAMESPACE_TOPIC) || child.childOf(VERSION_CONFIG_KEY)
-                    || child.childOf(POSIX_USER_KEY)) {
+                    || child.childOf(POSIX_USER_KEY) || child.childOf(WINDOWS_USER_KEY)) {
```

Comment:
shall we just use childOf the runwith namespace instead of specifying posix/windows?

Reply:
Child of runWith also has resource limit. That is handled separately on L119

Reply:
could do childof runwith && !childof system limits?

Reply:
This has a small issue where if the customer has set both Windows and Posix user in their deployment config and change one of them which is not applicable to the platform the deployment executes on, it will still end up restarting components, it would be nice to have this only react to the user config that it cares about